src/macros.js

```diff
@@ -6,10 +6,39 @@
 
 import symbols from "./symbols";
 import utils from "./utils";
+import {Token} from "./Token";
 
-// This function might one day accept additional argument and do more things.
-function defineMacro(name: string, body: string | () => string) {
-    module.exports[name] = body;
+/**
+ * Provides context to macros defined by functions. Implemented by
+ * MacroExpander.
+ */
+export interface MacroContextInterface {
+    /**
+     * Returns the topmost token on the stack, without expanding it.
+     * Similar in behavior to TeX's `\futurelet`.
+     */
+    future(): Token;
+
+    /**
+     * Expand the next token only once (if possible), and return the resulting
+     * top token on the stack (without removing anything from the stack).
+     * Similar in behavior to TeX's `\expandafter\futurelet`.
+     */
+    expandAfterFuture(): Token;
+}
+
+/** Macro tokens (in reverse order). */
+export type ParsedExpansion = {expansion: Token[], numArgs: number};
```

Comment:
Now that we've given this thing a name, `ParsedExpansion`, it might make sense to change it to be
```
{tokens: Token[], numArgs: number}
```
then
```
const {expansion: exp, numArgs} = this._getExpansion(name);
```
will become
```
const {tokens, numArgs} = this._getExpansion(name);
```
which is less awkward.

Reply:
Done.